### PR TITLE
refactor: Created Operation enumeration

### PR DESCRIPTION
**Pull request by EdwardBarclay**
TIS21-1032

### CODE REVIEW

File: src/test/java/uk/nhs/hee/tis/trainee/sync/service/ReferenceSyncServiceTest.java [hunk 1]
```diff
@@ -74,22 +76,12 @@ void shouldNotSyncRecordWhenTableNotSupported() {
     verifyNoInteractions(restTemplate);
   }
 
-  @Test
-  void shouldNotSyncRecordWhenOperationNotSupported() {
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/trainee/sync/service/ReferenceSyncServiceTest.java:78]: Can you create a test like this in the `RecordServiceTest` and it's at least technically possible that `record.operation` is null.

File: src/test/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncServiceTest.java [hunk 2]
```diff
@@ -111,23 +113,12 @@ void shouldNotSyncRecordWhenTableNotSupported() {
     verifyNoInteractions(restTemplate);
   }
 
-  @Test
-  void shouldNotSyncDetailsRecordWhenOperationNotSupported() {
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncServiceTest.java:115]: As above:

> Can you create a test like this in the RecordServiceTest and it's at least technically possible that record.operation is null.

**Review by JosephKelly — CHANGES_REQUESTED** [review]
Might be wrong but i would have thought the at least the mapper util should change

File: src/main/java/uk/nhs/hee/tis/trainee/sync/mapper/util/RecordUtil.java [hunk 3]
```diff
@@ -67,7 +68,7 @@ public String id(Map<String, String> data) {
 
   @Operation
   public String operation(Map<String, String> metadata) {
-    return metadata.get("operation");
+    return metadata.get("operation").toUpperCase();
```
[the remark this answers is not in the document]
Comment (Judge40) [reply src/main/java/uk/nhs/hee/tis/trainee/sync/mapper/util/RecordUtil.java:70]: The util is to get the values from the DTO and the DTO is string based. Mapstruct automatically maps from string to enum, but the casing has to match so we put this in here so the OOTB mapstruct mapping could work.

File: src/test/java/uk/nhs/hee/tis/trainee/sync/service/ReferenceSyncServiceTest.java (hunk 1 above)
Comment (Judge40) [reply src/test/java/uk/nhs/hee/tis/trainee/sync/service/ReferenceSyncServiceTest.java:78]: Done

File: src/test/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncServiceTest.java (hunk 2 above)
Comment (Judge40) [reply src/test/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncServiceTest.java:110]: Done

File: src/main/java/uk/nhs/hee/tis/trainee/sync/mapper/util/RecordUtil.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Judge40) [reply src/main/java/uk/nhs/hee/tis/trainee/sync/mapper/util/RecordUtil.java]: Refactored it to return either an `Operation` enum value or `IllegalArgumentException`

**Review by ReubenRobertsHEE — APPROVED** [review]
Looks good

**Review by JosephKelly — APPROVED** [review]

**Pull request merged**